substrate/primitives/core/src/ecdsa.rs

```diff
@@ -498,18 +477,6 @@ impl Pair {
 	}
 }
 
-// The `secp256k1` backend doesn't implement cleanup for their private keys.
-// Currently we should take care of wiping the secret from memory.
-// NOTE: this solution is not effective when `Pair` is moved around memory.
-// The very same problem affects other cryptographic backends that are just using
-// `zeroize`for their secrets.
-#[cfg(feature = "full_crypto")]
-impl Drop for Pair {
-	fn drop(&mut self) {
-		self.secret.non_secure_erase()
-	}
-}
-
```

Comment:
note: `Drop` shall be fine in [`k256::SigningKey`](https://docs.rs/ecdsa/0.16.9/src/ecdsa/signing.rs.html#380-384)